addons/payment_stripe/models/payment.py

```diff
@@ -149,14 +161,98 @@ def _get_feature_support(self):
         res['tokenize'].append('stripe')
         return res
 
+    def _handle_stripe_webhook(self, data):
+        """Process a webhook payload from Stripe.
+
+        Post-process a webhook payload to act upon the matching payment.transaction
+        record in Odoo.
+        """
+        SUPPORTED_WEBHOOKS = ['checkout.session.completed']
+        wh_type = data.get('type')
+        if wh_type not in SUPPORTED_WEBHOOKS:
+            _logger.info('unsupported webhook type %s, ignored', wh_type)
+            return False
+        else:
+            _logger.info('handling %s webhook event from stripe', wh_type)
+
+        signature = request.httprequest.headers.get('Stripe-Signature')
+        body = request.httprequest.data
+        self._verify_stripe_signature(signature, body)
+
+        stripe_object = data.get('data', {}).get('object')
+        if not stripe_object:
+            raise ValidationError(_('Stripe Webhook data does not conform to the expected API.'))
+        if wh_type == 'checkout.session.completed':
+            return self._handle_checkout_webhook(stripe_object)
+        return False
+
+    def _verify_stripe_signature(self, signature: str, body: bytes):
+        """
+        :param signature: t=xxx,v0=xxx,v1=xxx string provided in request header
+        :param body: bytes provided as the request body
+        :return: true if and only if signature matches hash of payload calculated with secret
+        :raises ValidationError: if signature doesn't match
+        """
+        sign_data = {k: v for (k, v) in [s.split('=') for s in signature.split(',')]}
+        event_timestamp = int(sign_data['t'])
+        tolerance_in_seconds = 600
+        if datetime.utcnow().timestamp() - event_timestamp > tolerance_in_seconds:
```

Comment:
FYI: `utcnow` and `now` are one and the same in odoo, the server is started in UTC (see `odoo/odoo/__init__.py:63`).

The `utcnow` can stay, it makes it obvious, i just saw an opportunity to give you some knowledge tidbit

Reply:
Ok, nice to know. I got burned by a badly configured MySQL server so I tend to pay extra attention to these kind of things now.